libsolidity/ast/AST.cpp

```diff
@@ -181,6 +181,13 @@ void ContractDefinition::setUserDocumentation(Json::Value const& _userDocumentat
 }
 
 
+std::string ContractDefinition::fullyQualifiedName() const
+{
+	std::string sourceString = *(location().sourceName);
```

Comment:
But that is actually a different thing. In the tests, the source file is not named. I would say that it is always possible to do a qualified lookup and in addition, you can do an unqualified lookup if the name is unique, which would guarantee backwards-compatibility.